dartagnan/src/main/java/com/dat3m/dartagnan/utils/dependable/DependencyGraph.java

```diff
@@ -225,4 +257,42 @@ public String toString() {
             return content.toString();
         }
     }
+
+    // -------------------------------------------------------------------
+    // -------- Original (recursive) strongConnect implementation --------
```

Comment:
We can get rid of this (and update the comment above referring to the recursive version)

Reply:
What do you mean by update the comment above? In what way? 
I left the recursive algorithm as "specification", cause it is hard to understand the non-recursive version on its own.

Reply:
I did not know you wanted to keep this comment as the specification of the code, and if this was removed, the comment referring to this had to be gone.

I am fine with keeping the commented code as specification, because I agree that the recursion-free version is not trivial.

Reply:
If I keep the original code with special comments that reference it, you can bet that this is intentional :P.
The reason I want to keep this is that in the off-chance that I need to change something in the future, I won't be able to understand the non-recursive version easily. To be fair, I don't even fully understand why the recursive version works, but at least I understand the recursive structure itself (i.e., that it is a DFS) :).